modules/mod-deep-learning/DeepModel.h

```diff
@@ -0,0 +1,117 @@
+/**********************************************************************
+
+   Audacity: A Digital Audio Editor
+   Audacity(R) is copyright (c) 1999-2021 Audacity Team.
+
+   DeepModel.h
+   Hugo Flores Garcia
+
+******************************************************************/
+/**
+
+\class DeepModel
+\brief base class for handling torchscript models
+
+*/
+/*******************************************************************/
+
+#pragma once
+
+#include "ModelCard.h"
+#include "wx/log.h"
+
+#include <tuple>
+
+#include <torch/script.h>
+#include <torch/torch.h>
+#include "AudacityException.h"
+#include "Identifier.h"
+
+class DeepModel;
+
+using ModulePtr = std::unique_ptr<torch::jit::script::Module>;
+using DeepModelHolder = std::shared_ptr<DeepModel>;
+using TensorWithTimestamps = std::tuple<torch::Tensor, torch::Tensor>;
+
+class ModelException final : public MessageBoxException
+{
+public:
+   ModelException(const TranslatableString msg, std::string trace) :
+                  m_msg(msg),
+                  m_trace(std::move(trace)),
+                  MessageBoxException{
+                     ExceptionType::Internal,
+                     XO("Deep Model Error")
+                  }
+   { 
+      if (!m_trace.empty()) 
+         wxLogError(wxString(m_trace)); 
+   }
+
+   // internal message
+   virtual const char* what() const throw () 
+      { return m_msg.Translation().c_str(); }
+
+   // user facing message
+   virtual TranslatableString ErrorMessage() const
+      { return XO("Deep Model Error: %s").Format(m_msg);}
+   
+   const TranslatableString m_msg;
+   const std::string m_trace;
+};
+
+class DeepModel final
+{
+public:
+   // base directory for deep learning models
+   // TODO: maybe we want to support a couple of search paths?
+   static FilePath DLModelsDir();
+   static FilePath BuiltInModulesDir();
+
+   DeepModel() = default;
+   DeepModel(ModelCard &card);
+
+   // @execsafety: strong 
```

Comment:
I only now noticed a systematic mistake of yours in Doxygen comments.

Please read this: https://www.doxygen.nl/manual/docblocks.html#specialblock

Your comment blocks are not correct, and none of your intended documentation comments for methods are going into the generated html pages.  (You can generate those for yourself if you install doxygen and run at the project root: `doxygen nographs.dox`)

To make the extraction work, you need to start line comments with
`//!`
Or you can make long comments with
```
/*!
...
*/
```

Reply:
fixed in 8cce7502a